pkg/kubelet/metrics/metrics.go

```diff
@@ -40,13 +43,31 @@ const (
 	VolumeStatsInodesKey         = "volume_stats_inodes"
 	VolumeStatsInodesFreeKey     = "volume_stats_inodes_free"
 	VolumeStatsInodesUsedKey     = "volume_stats_inodes_used"
-	// Metrics keys of remote runtime operations
+	// Metric keys for remote runtime operations
 	RuntimeOperationsKey        = "runtime_operations"
 	RuntimeOperationsLatencyKey = "runtime_operations_latency_microseconds"
 	RuntimeOperationsErrorsKey  = "runtime_operations_errors"
-	// Metrics keys of device plugin operations
+	// Metric keys for device plugin operations
 	DevicePluginRegistrationCountKey = "device_plugin_registration_count"
 	DevicePluginAllocationLatencyKey = "device_plugin_alloc_latency_microseconds"
+	// Metric keys for node config sources
+	// AssignedNodeConfigSourceKey's metric reports the config source the Kubelet believes is currently assigned to it
+	AssignedNodeConfigSourceKey = "assigned_node_config_source"
+	// AssignedNodeConfigSourceOkayKey's metric indicates whether the currently assigned config source and the config it contains is valid
+	AssignedNodeConfigSourceOkayKey       = "assigned_node_config_source_okay"
+	AssignedNodeConfigSourceStatusTrue    = 1
+	AssignedNodeConfigSourceStatusUnknown = 0
+	AssignedNodeConfigSourceStatusFalse   = -1
+	// LastKnownGoodNodeConfigSourceKey's metric indicates what the Kubelet is treating as it's last-known-good config source
+	LastKnownGoodNodeConfigSourceKey = "last_known_good_node_config_source"
```

Comment:
Try to make these hierarchical, that helps, e.g. `dynamic_config_last_known_source`